hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java

```diff
@@ -172,12 +172,15 @@ public static SortedMap<String, QueryDataType> resolveCompact(@Nonnull Schema sc
     public static QueryDataType resolveType(@Nonnull FieldKind kind) {
         switch (kind) {
             case BOOLEAN:
+            case NULLABLE_BOOLEAN:
                 return QueryDataType.BOOLEAN;
 
             case INT8:
+            case NULLABLE_INT8:
                 return QueryDataType.TINYINT;
 
             case INT16:
+            case NULLABLE_INT16:
                 return QueryDataType.SMALLINT;
 
             case CHAR:
```

Comment:
does compact serialization support `Character`?

Reply:
apparently not - maybe they should?

Reply:
https://hazelcast.slack.com/archives/C01JU7ZJYGP/p1681836482255999